src/grow_worker/worker.py

```diff
@@ -75,10 +80,45 @@ def run_mesido(input_esdl: str) -> str:
             update_progress_function=None,
             profile_reader=InfluxDBProfileReader,
         )
+        esdl_str = cast(str, solution.optimized_esdl_string)
+        # TODO get esdl_messages from solution after mesido update.
+        esdl_messages = []
+    except MesidoAssetIssueError as mesido_issues_error:
+        esdl_str = None
+        esdl_messages = parse_mesido_esdl_messages(
+            mesido_issues_error.general_issue, mesido_issues_error.message_per_asset_id
+        )
     except SystemExit as e:
         raise EarlySystemExit(e)
 
-    return cast(str, solution.optimized_esdl_string)
+    return esdl_str, esdl_messages
+
+
+def parse_mesido_esdl_messages(
+    general_message: str, object_messages: Dict[str, str]
+) -> List[EsdlMessage]:
+    """Convert mesido messages to a list of esdl messages in omotes format.
+
+    :param general_message: general message (not related to a specific ESDL object).
+    :param object_messages: esdl object messages per object id.
+    :return: list of EsdlMessage protobuf objects.
+    """
+    # TODO get severity from esdl message and add list of general messages after mesido update.
```

Comment:
The mesido team does not return the severity level from their package yet, so we can only assign an ERROR level at the moment, is that right?

Reply:
yes, that will come.